src/index/CompressedRelation.cpp

```diff
@@ -623,10 +631,18 @@ DecompressedBlock CompressedRelationReader::readPossiblyIncompleteBlock(
   AD_CORRECTNESS_CHECK(ADDITIONAL_COLUMN_GRAPH_ID <
                        blockMetadata.offsetsAndCompressedSize_.size());
 
-  // We first scan the complete block including ALL columns.
+  bool manuallyDeleteGraphColumn = scanConfig.graphFilter_.deleteGraphColumn_;
+  // We first scan the complete block including ALL columns with the following
+  // exception: If `manuallyDeleteGraphColumn` is true, then the `graphColumn`
+  // is contained inside `scanConfig.scanColumns`, but this function is supposed
+  // to delete it. In this case we will not scan the graph column, that way the
+  // `readAndDecompressBlock` function will correctly delete all duplicates from
+  // the block. The downside of this approach is that further down we have to be
+  // aware of this already dropped column when assembling the final result.
   std::vector<ColumnIndex> allAdditionalColumns;
   ql::ranges::copy(
-      ql::views::iota(ADDITIONAL_COLUMN_GRAPH_ID,
+      ql::views::iota(ADDITIONAL_COLUMN_GRAPH_ID +
+                          static_cast<size_t>(manuallyDeleteGraphColumn),
```

Comment:
Off-topic: But I'm convinced that half of QLever breaks if `ADDITIONAL_COLUMN_GRAPH_ID` ever was changed to a number that's not 3.